lib/BridgedRoom.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+function BridgedRoom(bridge, gitter, gitterRealtime, matrixRoomModel, gitterRoomModel) {
+  this._bridge = bridge;
+  this._gitter = gitter;
+  this._gitterRealtime = gitterRealtime;
+  this._matrixRoomModel = matrixRoomModel;
+  this._gitterRoomModel = gitterRoomModel;
+}
+
+BridgedRoom.prototype.matrixRoomId = function() {
+  return this._matrixRoomModel.getId();
+};
+
+BridgedRoom.prototype.gitterRoomName = function() {
+  return this._gitterRoomModel.getId();
+};
+
+BridgedRoom.prototype.joinAndStart = function() {
+  // We have to find out our own gitter user ID so we can ignore reflections of
+  // messages we sent
+  return this._bridge.getMyGitterUserId().then((gitterUserId) => {
+    this._gitterUserId = gitterUserId;
+
+    return this._gitter.rooms.join(this.gitterRoomName()).then((room) => {
```

Comment:
You can just return this promise and `then` on the result to stop the indent levels stacking up:

```
    return this._gitter.rooms.join(this.gitterRoomName());
}).then((room) => {
    [etc]
}
```